### PR TITLE
Add additional properties to the Block Editor activation events

### PR DESCRIPTION
**Pull request by daniloercoli**
Add additional properties to  `gutenberg_enabled` , and `gutenberg_disabled` analytics events:
The property `source` can now have the following values:
- via-site-settings
- on-block-post-open
- on-site-creation

Test 1:
- Flip the block editor setting in Site settings and make sure the property is added correctly

Test 2:
- Create a new site, and make sure the property is added with the correct value
- Tap on new post, and make sure nothing is tracked

Test 3:
- Clear your editor settings on the web
- Re-install the app
- open a block based post
- the popup should appear, and the correct property value bumped with analytics

cc @etoledom Since we need the iOS counterpart of this PR

Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by hypest** [comment]
Let's target the 13.0 release branch for this one and the WPiOS one when it gets opened? cc @etoledom and headsup @loremattei .

**Comment by daniloercoli** [comment]
> Let's target the 13.0 release branch for this one and the WPiOS one when it gets opened? cc @etoledom and headsup @loremattei .

I wanted to target 13.0 branch, but it's still not there :)

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -1550,10 +1551,17 @@ private void showGutenbergInformativeDialog() {
     }
 
     private void setGutenbergEnabledIfNeeded() {
+        boolean newSitePopupNeeded = AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl());
         if ((TextUtils.isEmpty(mSite.getMobileEditor()) && !mIsNewPost)
-            || AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl())) {
+            || newSitePopupNeeded) {
             SiteUtils.enableBlockEditor(mDispatcher, mSite);
-            AnalyticsUtils.trackWithSiteDetails(Stat.EDITOR_GUTENBERG_ENABLED, mSite);
+            Map<String, Object> properties = new HashMap<>();
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Nit but, can we move the whole line `Map<String, Object> properties = new HashMap<>();` inside the conditional to be closer to the code using it? We don't need it outside, right?

While at it, and certainly not a blocker, WDYT about introducing a helper method to the enum object to quickly return a prefilled properties map? Something like:

```
public enum BlockEditorEnabledSource {
    VIA_SITE_SETTINGS,
    ON_SITE_CREATION,
    ON_BLOCK_POST_OPENING;

    public Map<String, Object> asPropertyMap() {
        Map<String, Object> properties = new HashMap<>();
        properties.put("source", name());
        return properties;
    }
}
```

We can then for example use it as `BlockEditorEnabledSource.ON_BLOCK_POST_OPENING.asPropertyMap()`.

Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/10300/commits/5809aa71e382401a7b9a19bb5fc277d2f51bac93

**Comment by hypest** [comment]
Hey Danilo, saw 6c2f05a but I wonder if it will make it hard for us to understand the tracks if we don't set a value there. Like, when that case happens that we don't still have the setting at hand, how will we interpret the tracks we see? WDYT? 🤔

**Comment by daniloercoli** [comment]
> Hey Danilo, saw [6c2f05a](https://github.com/wordpress-mobile/WordPress-Android/commit/6c2f05a46c89f129803956cdeec1924a57416c8a) but I wonder if it will make it hard for us to understand the tracks if we don't set a value there. Like, when that case happens that we don't still have the setting at hand, how will we interpret the tracks we see? WDYT? 🤔

Well, right after the login we don't have the info to bump the property IS_GUTENBERG_ENABLED correctly, since sites info are still not there. I think its's better to not add a wrong value (`false`) and do not pass it altogether. The correct value will be bump later.  I think we can refresh metadata when the editor info are returned from the server, even though it could increase the calls to `refreshMetadata`.

It should not be a big difference, backend query side, since we can read the most recent event from the Track and check if the user has GB on or off. 

On a side note, there is no user related GB setting anymore, wondering if we need to keep this property at user level...can't we move it to the general properties added when tracking events with site details `trackWithSiteDetails`?

**Comment by jkmassel** [comment]
I'm bumping this to `13.1` as part of the `13.0` code freeze.

If you'd like it to be released as part of `13.0`, please merge it against `release/13.0` then DM me, and I'll be happy to issue a new beta release! :)

File: WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java [hunk 2]
```diff
@@ -110,8 +122,8 @@ public static void refreshMetadata(AccountStore accountStore, SiteStore siteStor
         if (siteLocalId != -1) {
             // Site previously selected, use it
             SiteModel selectedSite = siteStore.getSiteByLocalId(siteLocalId);
-            // If saved site exist, then add info
-            if (selectedSite != null) {
+            // If saved site exist, then add editor info
+            if (selectedSite != null && !TextUtils.isEmpty(selectedSite.getMobileEditor())) {
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java]: Apparently ([see here](https://github.com/wordpress-mobile/WordPress-Android/blob/b3af4aeeaa176902280352ceb70860c7daae5bfa/libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java#L527) that the flag is set in the user section) we'll write `false` by default even if we skip setting the flag here. Perhaps leaving empty is a better approach so, wdyt about introducing an additional flag in `AnalyticsMetadata` to denote if we have set the flag at `AnalyticsMetadata.mIsGutenbergEnabled` at all or not?

Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java]: Addressed in b59bd9126709f8a11280f2105fc5398924c2fa02

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 3]
```diff
@@ -1550,10 +1551,17 @@ private void showGutenbergInformativeDialog() {
...
+        boolean newSitePopupNeeded = AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl());
         if ((TextUtils.isEmpty(mSite.getMobileEditor()) && !mIsNewPost)
-            || AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl())) {
+            || newSitePopupNeeded) {
             SiteUtils.enableBlockEditor(mDispatcher, mSite);
-            AnalyticsUtils.trackWithSiteDetails(Stat.EDITOR_GUTENBERG_ENABLED, mSite);
+
+            if (!newSitePopupNeeded) {
+                // Don't track the activation of the block editor on new sites.
+                // The flip to GB happened at site creation time, and the flip is tracked there with proper details
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Hmm, should the `SiteUtils.enableBlockEditor(mDispatcher, mSite);` (line number 1557 above) be inside this conditional too then? 🤔

Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Well, I think it's better to keep it out, since the user can change the option back to Aztec from settings right after the site creation. Even before creating/opening a new post. It's an edge case I know, but better to show both the dialog and repeat the network call to the server. It's just a call, that doesn't change anything on new sites 99% of the cases.

**Comment by hypest** [comment]
> Test 1:
> - Flip the block editor setting in Site settings and make sure the property is added correctly
>
> Test 2:
> - Create a new site, and make sure the property is added with the correct value
> - Tap on new post, and make sure nothing is tracked
>
> Test 3:
> - Clear your editor settings on the web
> - Re-install the app
> - open a block based post
> - the popup should appear, and the correct property value bumped with analytics

Verified that all test cases worked fine for me 👍

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 3 above)
Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: As per chat convo - cleaned up the code an made more easy to follow in https://github.com/wordpress-mobile/WordPress-Android/pull/10300/commits/4db82286865a7cfc3b0c21c8e868ccd757db112e
